v4/service.go

```diff
@@ -0,0 +1,81 @@
+package suture
+
+import (
+	"context"
+	"errors"
+)
+
+/*
+Service is the interface that describes a service to a Supervisor.
+
+Serve Method
+
+The Serve method is called by a Supervisor to start the service.
+The service should execute within the goroutine that this is
+called in, that is, it should not spawn a "worker" goroutine.
+If this function either returns error or panics, the Supervisor
+will call it again, unless it returns special error sentinel values
+
+A Serve method SHOULD do as much cleanup of the state as possible,
+to prevent any corruption in the previous state from crashing the
+service again.
+
+The Supervisor will pass a ctx down to the service. That context
```

Comment:
It would be helpful to explain the intended use of the context. This only tells me what _not_ to do.